py-polars/polars/interchange/from_dataframe.py

```diff
@@ -37,41 +44,277 @@ def from_dataframe(df: SupportsInterchange, *, allow_copy: bool = True) -> DataF
             f"`df` of type {type(df).__name__!r} does not support the dataframe interchange protocol"
         )
 
-    pa_table = _df_to_pyarrow_table(df, allow_copy=allow_copy)
-    return from_arrow(pa_table, rechunk=allow_copy)  # type: ignore[return-value]
+    return _from_dataframe(
+        df.__dataframe__(allow_copy=allow_copy),  # type: ignore[arg-type]
+        allow_copy=allow_copy,
+    )
 
 
-def _df_to_pyarrow_table(df: Any, *, allow_copy: bool = False) -> pa.Table:
-    if not _PYARROW_AVAILABLE or parse_version(pa.__version__) < (11, 0):
-        raise ImportError(
-            "pyarrow>=11.0.0 is required for converting a dataframe interchange object"
-            " to a Polars dataframe"
+def _from_dataframe(df: InterchangeDataFrame, *, allow_copy: bool = True) -> DataFrame:
+    chunks = []
+    for chunk in df.get_chunks():
+        polars_chunk = _protocol_df_chunk_to_polars(chunk, allow_copy=allow_copy)
+        chunks.append(polars_chunk)
+
+    # Handle implementations that incorrectly yield no chunks for an empty dataframe
+    if not chunks:
+        polars_chunk = _protocol_df_chunk_to_polars(df, allow_copy=allow_copy)
+        chunks.append(polars_chunk)
+
+    return F.concat(chunks, rechunk=False)
+
+
+def _protocol_df_chunk_to_polars(
+    df: InterchangeDataFrame, *, allow_copy: bool = True
+) -> DataFrame:
```

Comment:
this is a really small nitpick, but for internal functions, how about not setting defaults? so, just `*, allow_copy: bool)`. Otherwise, it's easy to forget to set it when calling then function (at least, that's something I often do)

Reply:
That's fair enough. Will update 👍